frontend/components/Archives.tsx

```diff
@@ -42,186 +39,232 @@ export interface IFetchArchives {
     startFrom: number,
     amount: number,
     setArchives: React.Dispatch<React.SetStateAction<IArchive[]>>,
-    setLoadedIndices: React.Dispatch<React.SetStateAction<number>>,
+    setLastLoadedIndex: React.Dispatch<React.SetStateAction<number>>,
     ownerProfiles: { [key: string]: any },
     setOwnerProfiles: React.Dispatch<
       React.SetStateAction<{ [key: string]: any }>
     >,
   ): Promise<void>;
 }
 
+export interface IFetchArchivesCount {
+  (
+    setArchivesCount: React.Dispatch<React.SetStateAction<number | undefined>>,
```

Comment:
can this be undefined? it seems there will always be a number

Reply:
@democrazyfi starts off undefined. that way we cant easily track that result is still loading